dev/com.ibm.ws.cdi.weld/src/com/ibm/ws/cdi/liberty/CDIRuntimeImpl.java

```diff
@@ -421,9 +425,11 @@ public void applicationStarting(ApplicationInfo appInfo) throws StateChangeExcep
                 this.runtimeFactory.removeApplication(appInfo);
                 return;
             }
-            newCL = getRealAppClassLoader(application);
 
-            if (newCL != null) {
+            ClassLoader appCL = getRealAppClassLoader(application);
+            if (appCL != null) {
+                newCL = classLoadingSRRef.getServiceWithException().createThreadContextClassLoader(appCL);
```

Comment:
It looks like this could be the first call to create the TCCL, so destroying it in the finally block might really destroy it, and therefore keeping a reference to it to destroy it when the application is stopped is the safest thing to do.